joomla/joomla-cms
# [4.0] Improvement Vote Plugin

Pull Request for Issue #30193.

### Summary of Changes

In the current situation only two stars can be seen.
In the new situation, a half star has been added.
Font Awesome icons are used and no longer two (bad) .png images.


### Testing Instructions

Install Multilingual Sample Data
Use `npm ci`  (npm run build:css) or `Prebuilt package` (bottom page)

I.
- Enable Vote Plugin
- Article Edit > Options > Show Voting
- Also create a menu link (list all categories) to an article.

Cast votes.

II.
Change the values in the table #__content_rating.
Values for testing: round = 0.5 (old round to 1)
rating 3.2 -> 3 stars 
rating 3.3 -> 3.5 stars (10 : 3 = rating_sum / rating_count)
rating 3.4 -> 3.5 stars
rating 3.7 -> 3.5 stars
rating 3.8 -> 4 stars

III.
Test in a LFT and a RTL language.

IV.
Test overrides (images).

V.
Code review + security.

The _role_ and _aria-label_ attribute have been added for accessibility. See the html source code too.

### Actual result BEFORE applying this Pull Request

Two stars can be seen: a whole and an empty star.
E.g. vote average = 3.3 becomes 3 stars.

### Expected result AFTER applying this Pull Request

Three different stars can be seen: a whole, a half and an empty star.
E.g. vote average = 3.3 becomes 3.5 stars.


![issue 31098](https://user-images.githubusercontent.com/9271775/96114417-e2c82800-0ee5-11eb-9478-b930e12afe7c.png)


### Documentation Changes Required

?

### Additional comment

When PR #31096 is added, there are more options in terms of the look and feel of the stars.

[EDIT]
